File: gdscript/src/main/java/gdscript/psi/impl/GdCastExImpl.java
package gdscript.psi.impl;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import com.intellij.psi.util.PsiTreeUtil;
import gdscript.psi.GdCastEx;
import gdscript.psi.GdExpr;
import gdscript.psi.GdTypedVal;
import gdscript.psi.GdVisitor;
import org.jetbrains.annotations.NotNull;

public class GdCastExImpl extends GdExprImpl implements GdCastEx {

  public GdCastExImpl(@NotNull ASTNode node) {
    super(node);
  }

  @Override
  public void accept(@NotNull GdVisitor visitor) {
    visitor.visitCastEx(this);
  }

  @Override
  public void accept(@NotNull PsiElementVisitor visitor) {
    if (visitor instanceof GdVisitor) accept((GdVisitor)visitor);
    else super.accept(visitor);
  }

  @Override
  @NotNull
  public GdExpr getExpr() {
    return notNullChild(PsiTreeUtil.getChildOfType(this, GdExpr.class));
  }

  @Override
  @NotNull
  public GdTypedVal getTypedVal() {
    return notNullChild(PsiTreeUtil.getChildOfType(this, GdTypedVal.class));
  }

}

File: gdscript/src/main/kotlin/gdscript/psi/impl/GdCastExImpl.kt
package gdscript.psi.impl

import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElementVisitor
import com.intellij.psi.util.PsiTreeUtil
import gdscript.psi.GdCastEx
import gdscript.psi.GdExpr
import gdscript.psi.GdTypedVal
import gdscript.psi.GdVisitor

class GdCastExImpl(node: ASTNode) : GdExprImpl(node), GdCastEx {
    override fun accept(visitor: GdVisitor) {
        visitor.visitCastEx(this)
    }

    override fun accept(visitor: PsiElementVisitor) {
        if (visitor is GdVisitor) accept(visitor)
        else super.accept(visitor)
    }

    override val expr: GdExpr
        get() = notNullChild(PsiTreeUtil.getChildOfType(this, GdExpr::class.java))

    override val typedVal: GdTypedVal
        get() = notNullChild(PsiTreeUtil.getChildOfType(this, GdTypedVal::class.java))
}
